client/src/components/SandwichShow.js

```diff
@@ -98,10 +99,53 @@ const SandwichShow = (props) => {
     );
   });
 
+  const roundHalf = (num) => {
+    return Math.round(num*2)/2
+  }
+
+  let rounded = roundHalf(sandwich.averageRating)
+
+  let roundedArray = []
+  if ((rounded/1) >= 1) {
+    roundedArray.push(1)
+  } 
+  if ((rounded/2) >= 1) {
+    roundedArray.push(2)
+  }
+  if ((rounded/3) >= 1) {
+    roundedArray.push(3)
+  }
+  if ((rounded/4) >= 1) {
+    roundedArray.push(4)
+  }
+  if ((rounded/5) >= 1) {
+    roundedArray.push(5)
+  }
+  if ((rounded%1) === .5) {
+    roundedArray.push(.5)
+  }
```

Comment:
this logic is just too ripe for abstraction for me to let it pass. 

I want you to make a function that performs this logic. This function can exist in a `services` folder in `client`. This function should take in an argument of your `rounded` sandwich rating, and sput out an array with the number values you want. Then import the function and call it here!